src/compile/scale/range.ts

```diff
@@ -45,13 +45,19 @@ function parseUnitScaleRange(model: UnitModel) {
 
     // Read if there is a specified width/height
     const sizeType = channel === 'x' ? 'width' : channel === 'y' ? 'height' : undefined;
-    const specifiedSize = sizeType ? model.component.layoutSize.get(sizeType) : undefined;
+    let sizeSpecified = sizeType ? !!model.component.layoutSize.get(sizeType) : undefined;
+
+    // if autosize is fit, size cannot be data driven
+    if (sizeType && model.fit && !sizeSpecified) {
```

Comment:
If neither width/height or rangeStep is specified, we shouldn't throw warning.  

Basically, if user either specify width/height or fit, that means users intend to fit the range step.  Thus the warning would be confusing for users.   

I think we should instead
- [ ] update relevant docs (`size.md` + JSDocs for range step and width/height) to explain about this.
- [ ] throw warning only if users specify both rangeStep and fit.

Reply:
Also, currently the logic here throw warning even for height when y is continuous. 

![image](https://user-images.githubusercontent.com/111269/30258602-a343c636-966f-11e7-9dc7-362e60c674ee.png)

(There are two of these warnings in the screenshot.)

Reply:
Good catch. Chrome just showed me a 2 in front of the message that I missed.